patchers/value/value.go

```diff
@@ -0,0 +1,260 @@
+// Use it directly as an Option to expr.Compile()
+var Patcher = func() expr.Option {
+	vPatcher := patcher{}
+	return func(c *conf.Config) {
+		c.Visitors = append(c.Visitors, vPatcher)
+		vPatcher.ApplyOptions(c)
+	}
+}()
+
+// A ExprValuer provides a generic function for a custom type to return standard go values.
+// It allows for returning a `nil` value but does not provide any type checking at expression compile.
+//
+// A custom type may implement both ExprValuer and a type specific interface to enable both
+// compile time checking and the ability to return a `nil` value.
+type ExprValuer interface {
+	ExprValue() any
+}
+
+type IntValuer interface {
+	IntValue() int
+}
+
+type BoolValuer interface {
+	BoolValue() bool
+}
+
+type Int8Valuer interface {
+	Int8Value() int8
+}
+
+type Int16Valuer interface {
+	Int16Value() int16
+}
+
+type Int32Valuer interface {
+	Int32Value() int32
+}
+
+type Int64Valuer interface {
+	Int64Value() int64
+}
+
+type UintValuer interface {
+	UintValue() uint
+}
+
+type Uint8Valuer interface {
+	Uint8Value() uint8
+}
+
+type Uint16Valuer interface {
+	Uint16Value() uint16
+}
+
+type Uint32Valuer interface {
+	Uint32Value() uint32
+}
+
+type Uint64Valuer interface {
+	Uint64Value() uint64
+}
+
+type Float32Valuer interface {
+	Float32Value() float32
+}
+
+type Float64Valuer interface {
+	Float64Value() float64
+}
+
+type StringValuer interface {
+	StringValue() string
+}
+
+type TimeValuer interface {
+	TimeValue() time.Time
+}
+
+type DurationValuer interface {
+	DurationValue() time.Duration
+}
+
+type ArrayValuer interface {
+	ArrayValue() []any
+}
+
+type MapValuer interface {
+	MapValue() map[string]any
+}
+
+var supportedInterfaces = []reflect.Type{
+	reflect.TypeOf((*ExprValuer)(nil)).Elem(),
+	reflect.TypeOf((*BoolValuer)(nil)).Elem(),
+	reflect.TypeOf((*IntValuer)(nil)).Elem(),
+	reflect.TypeOf((*Int8Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Int16Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Int32Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Int64Valuer)(nil)).Elem(),
+	reflect.TypeOf((*UintValuer)(nil)).Elem(),
+	reflect.TypeOf((*Uint8Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Uint16Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Uint32Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Uint64Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Float32Valuer)(nil)).Elem(),
+	reflect.TypeOf((*Float64Valuer)(nil)).Elem(),
+	reflect.TypeOf((*StringValuer)(nil)).Elem(),
+	reflect.TypeOf((*TimeValuer)(nil)).Elem(),
+	reflect.TypeOf((*DurationValuer)(nil)).Elem(),
+	reflect.TypeOf((*ArrayValuer)(nil)).Elem(),
+	reflect.TypeOf((*MapValuer)(nil)).Elem(),
+}
+
+type patcher struct{}
+
+func (patcher) Visit(node *ast.Node) {
+	id, ok := (*node).(*ast.IdentifierNode)
+	if !ok {
+		return
+	}
+
+	callnode := &ast.CallNode{
+		Callee:    &ast.IdentifierNode{Value: "getExprValue"},
+		Arguments: []ast.Node{id},
+	}
+
+	nodeType := id.Type()
+
+	for _, t := range supportedInterfaces {
+		if nodeType.Implements(t) {
```

Comment:
What going to be if the type `t` is nil?

Reply:
`t` should never be nil.

`supportedInterfaces` is a slice containing all supported interface types and should never contain a `nil`.